pkg/proxy/iptables/proxier.go

```diff
@@ -423,16 +425,24 @@ func (proxier *Proxier) SyncLoop() {
 	}
 }
 
-// OnServiceUpdate tracks the active set of service proxies.
-// They will be synchronized using syncProxyRules()
-func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
-	start := time.Now()
-	defer func() {
-		glog.V(4).Infof("OnServiceUpdate took %v for %d services", time.Since(start), len(allServices))
-	}()
-	proxier.mu.Lock()
-	defer proxier.mu.Unlock()
-	proxier.haveReceivedServiceUpdate = true
+type healthCheckPort struct {
+	namespace types.NamespacedName
+	nodeport  int
+	// if true, should be added to healthcheck listener service.  If false,
+	// shoul dbe deleted.
```

Comment:
typo
(also, kind of an awkward API. Maybe return separate add/remove arrays rather than a single array of adds-and-removes?)